Carlos/NetworkFetcher.swift

```diff
@@ -65,15 +65,42 @@ public class NetworkFetcher: CacheLevel {
       }
     }
     
-    private func failWithCode(code: NetworkFetcherError, failure fail : ((NSError?) -> ())) {
+    private func failWithCode(code: NetworkFetcherError, failure fail : ((NSError?,Request) -> ())) {
       dispatch_async(dispatch_get_main_queue()) {
-        fail(errorWithCode(code.rawValue))
+        fail(errorWithCode(code.rawValue),self)
       }
     }
   }
-  
-  private var pendingRequests: [String: Request] = [:]
-  
+
+  private lazy var lockQueue: dispatch_queue_t = {
+    return dispatch_queue_create(CarlosGlobals.QueueNamePrefix + "NetworkFetcher", nil)
+  }()
+
+  private var pendingRequests: [Request] = []
+
+  private func addPendingRequest(request: Request) {
+    dispatch_async(lockQueue) {
+      self.pendingRequests.append(request)
+    }
+  }
+
+  private func removePendingRequests(request: Request) {
+    dispatch_async(lockQueue) {
+      var idx: Int?
+
+      for (index, obj) in enumerate(self.pendingRequests) {
```

Comment:
isn't there any way to find the index via the "find" function we have in Swift? or something less low-level?